### DISCUSSION TITLE
I'm facing some missing extensions

### DISCUSSION THREAD

**Discussion by marcosfreitas** [mlocati/docker-php-extension-installer#602] [Q&A] [answered]
Hi Guys, thanks for your amazing work on this project.

I have an image based on `php7.4-fpm-alpine` that uses this script to install some extensions. By default, the GD and Imagick are installed.

Today I got some errors when handling a composer update and to my surprise:

- 🐛  the gd ext is showing as not installed, but `php -m` lists it.
- 🐛  there is a php8 located at /etc an I don't know why

I wondering how this php8 goes there and if it is causing some conflict in compose.

- 👎🏻 I'd check the base image but there doesn't exist any php8 binary installed.
- 👎🏻 I rebuilt my image but the problem continues and I have no instructions to work with PHP 8.

So, basically, I have two PHP versions installed and some missing extensions. Any Ideas?

My image:
![image](https://user-images.githubusercontent.com/1065688/174465216-0a606605-3e28-428b-991e-6b170d953e70.png)
[https://github.com/Skinny-Hosts/core-php-fpm-docker-image/blob/master/Dockerfile](https://github.com/Skinny-Hosts/core-php-fpm-docker-image/blob/master/Dockerfile)

I've some descriptions [here](https://github.com/Skinny-Hosts/core-php-fpm-docker-image/issues/10)

Alpine base image:
![image](https://user-images.githubusercontent.com/1065688/174465264-5df547a9-b5ff-45c7-8e5f-05afa3b5a4f7.png)

**Comment by marcosfreitas** [comment]
Seems that running this command makes that php8 has been installed. Without it, php7 version remains.

```
apk add --no-cache shadow bash composer &&\
composer self-update --2
```

I'm still investigating about the missing extensions conflict.

**Comment by mlocati** [comment, marked as answer]
Maybe installing composer with apk causes apk to install PHP? And that's absolutely wrong: in the docker php images, PHP is compiled from source: you must not install it (or any PHP extension, or any package that depends on PHP) with apk.
The solution? Don't install composer with apk, and read https://github.com/mlocati/docker-php-extension-installer#installing-composer

**Reply by marcosfreitas** [reply]
Yeah, you are right. I saw this capability yesterday reading the docs but I don't remember to saw it before.

The curious part is that the composer is being installed this way a long time ago and I didn't have this problem before.

I think I was getting an outdated version of something... never mind.

I'm changing the installation and I will put a status here.
